src/components/Extensions/Details/DetailCard/Tabs/ParamList.tsx

```diff
@@ -22,11 +22,110 @@ import { Callout } from '../../../../common/Callout';
 import { DOCS_BASE } from '../../../../common/links/DocsLink';
 import { Markdown } from '../../../../common/Markdown';
 import { useExtension } from '../../../api/useExtension';
+import { ParamType } from '../../../models';
 import styles from './ParamList.module.scss';
 import { ParamValue } from './ParamValue';
 
 function ParamList() {
   const extension = useExtension()!;
+  let eventsByPrefix: Map<string, string[]> = new Map<string, string[]>(); // maps prefix ==> list of types
+  let eventDescriptionMapping: Map<string, string> = new Map<string, string>(); // maps type ==> description
+  let allowedEventTypesDiv;
+  let channelLocationDiv;
+  if (!!extension.events) {
+    extension.events.forEach((event) => {
+      eventDescriptionMapping.set(event.type, event.description);
+    });
+  }
+  if (!!extension.allowedEventTypes && extension.allowedEventTypes.length > 0) {
+    extension.allowedEventTypes.forEach((eventType) => {
+      const prefix = eventType.split('.').slice(0, -1).join('.');
+      if (!eventsByPrefix.get(prefix)) {
+        eventsByPrefix.set(prefix, [eventType]);
+      } else {
+        eventsByPrefix.get(prefix)?.push(eventType);
+      }
+    });
```

Comment:
I'm pretty sure I've seen lodash int he codebase, coould we just use keyBy/groupby here?

Reply:
Can you elaborate a bit more on this one? I can see how the keyBy would work to generate eventsByPrefix. But where does the groupBy fit in?

Reply:
allowedEventTypes - this seems like what groupBy does?